lib/App.ts

```diff
@@ -1,25 +1,34 @@
-import Component from "./Component.js";
+import _ from "lodash";
 import Navigo from "navigo";
+import AppOptions from "./AppOptions.js";
+import Component from "./Component.js";
 
 /**
  * This class creates a page component for each route in the application and initializes it and all its children.
  */
 export default class App {
+    private isAuthed = false; // whether the base page was loaded with ?isAuthed
     private pages: {[route: string]: Component} = {}; // default assign so we can push to it
     private router: Navigo;
     private basePath: string; // path that prefixes all components
-    private routeSource: string; // endpoint returning a list of the app's routes
+    public readonly options: AppOptions;
 
-    public constructor(routeSource?: string) {
+    public constructor(options?: AppOptions) {
+        this.options = _.defaultsDeep(options, {
+            auth: {
+                loginUrl: "/login",
+                logoutUrl: "/logout"
+            },
+            routeSource: "/delta/v1/routes"
+        } as AppOptions);
         this.basePath = window.location.pathname;
         this.router = new Navigo();
-        // if no route source is provided, use default endpoint in eta-web-delta module
-        this.routeSource = routeSource || "/delta/v1/routes";
+        this.isAuthed = window.location.search.includes("isAuthed");
```

Comment:
I agree that it's not an ideal solution, but the alternative is potentially impossible, since there are several different types of login possible (CAS, local, etc). This is a viable solution for now.